### PR TITLE
Fix polar facecolor

### LINKED ISSUE #5996: 2.x: Figure.add_axes(..., facecolor='color') does not set axis background colour
The [polar legend demo](http://matplotlib.org/devdocs/examples/pylab_examples/polar_legend.html) sets the face colour using the `facecolor=` keyword argument, but the background is still white.
polar legend demo
It still works correctly if you pass the `axisbg=` keyword argument, but then that raises a deprecation warning.

### PR DESCRIPTION
**Pull request by tacaswell**
Closes #5996

### CODE REVIEW

File: lib/matplotlib/axes/_base.py [hunk 1]
```diff
@@ -508,13 +509,17 @@ def __init__(self, fig, rect,
 
         # this call may differ for non-sep axes, e.g., polar
         self._init_axis()
-
-        if axisbg is None:
-            axisbg = rcParams['axes.facecolor']
-        else:
+        if axisbg is not None and facecolor is not None:
+            raise TypeError('Both axisbg and facecolor are not None. '
+                            'These keywords are aliases, only one maybe '
```
Comment (efiring) [inline lib/matplotlib/axes/_base.py]: maybe -> may be

File: lib/matplotlib/axes/_base.py [hunk 2]
```diff
@@ -2722,7 +2727,7 @@ def set_axis_bgcolor(self, color):
         warnings.warn(
```
Comment (efiring) [inline lib/matplotlib/axes/_base.py]: Is this warning still needed, given the decorator?

Comment (tacaswell) [inline lib/matplotlib/axes/_base.py]: I think you are correct.

**Comment by tacaswell** [comment]
Also noticed that we do not even really need that private variable, so fully removed it.

**Comment by mdboom** [comment]
:+1:  I think this may already be on master, though: 5f1127c4122473d114e9cbf7557de29e74f3752f

**Comment by tacaswell** [comment]
well, drat.

**Comment by tacaswell** [comment]
This PR looks like it touches a bit more than #5501 does.  This adds face_color as an explicit kwarg that the `_BaseAxes.__init__` pulls off rather than letting it fall all the way through to the `update` method.

I need to rip out half of the last commit.

**Comment by tacaswell** [comment]
There is a sutble API difference between `set_axis_bgcolor` and `set_facecolor`, the later (used to) set the internal `self._facecolor` (`self._axisbg`) attribute which is what `cla()` uses to restore the color.

**Pull request merged**